container-engine-lib/lib/backend_impls/docker/docker_manager/docker_auth_test.go

```diff
@@ -0,0 +1,97 @@
+package docker_manager
+
+import (
+	"encoding/base64"
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+// writeStaticConfig writes a static Docker config.json file to a temporary directory
+func writeStaticConfig(t *testing.T, configContent string) string {
+	tmpDir, err := os.MkdirTemp("", "docker-config")
+	if err != nil {
+		t.Fatalf("Failed to create temp directory: %v", err)
+	}
+
+	configPath := tmpDir + "/config.json"
+	err = os.WriteFile(configPath, []byte(configContent), 0600)
+	if err != nil {
+		t.Fatalf("Failed to write config.json: %v", err)
+	}
+
+	// Set the DOCKER_CONFIG environment variable to the temp directory
+	os.Setenv("DOCKER_CONFIG", tmpDir)
```

Comment:
Can we make `DOCKER_CONFIG` into a shared const variable in `docker_auth.go` since we're using it here and in `loadDockerAuth`?